src/io/timeout.rs

```diff
@@ -67,7 +67,7 @@ where
         let this = self.project();
         match this.future.poll(cx) {
             Poll::Pending => {}
-            other => return other,
+            other @ Poll::Ready(..) => return other,
```

Comment:
Unnecessary, if `Poll` were to expand there would be issues anyways and that would almost certainly have to happen in a rust edition.

Reply:
I think the goal here is to make clearer to readers that the only possibility is `Poll::Ready(..)`, rather than being unclear about the possibly multiple cases it covers.